spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/StatsdReporterTest.java

```diff
@@ -174,19 +180,150 @@ public void testTimers() throws Exception {
+            Meter meter = new Meter();
+            meter.mark();
+            meter.mark(2);
+            meters.put("test-app-01.driver.TestMeter", meter);
+
+            reporter.report(new TreeMap<>(), new TreeMap<>(), new TreeMap<>(), meters, new TreeMap<>());
+
+            sleep(100);
+
+            String prefix = "spark.driver.testmeter.";
+            assertThatNotExists(server, prefix, "count", "3|g");
+            assertThatExists(server, prefix, "m1_rate", "0.00|ms");
+            assertThatExists(server, prefix, "m5_rate", "0.00|ms");
+            assertThatExists(server, prefix, "m15_rate", "0.00|ms");
+            // mean value depends on the timing and can't be predicted for the test
+            assertThatExists(server, prefix, "mean_rate", "|ms");
+            assertEquals(4, server.receivedMessages().size());
+
+        }
+    }
+    
+    @Test
+    public void testAttributeIncludeFilter() throws Exception {
+        final int testUdpPort = 4448;
+        try (DatagramTestServer server = DatagramTestServer.run(testUdpPort)) {
+        	EnumSet<MetricAttribute> includesAttributes = EnumSet.copyOf(Arrays.asList(MetricAttribute.COUNT));
+        	StatsdReporterFactory reporterFactory = new StatsdReporterFactory();
+        	reporterFactory.setFormatter(formatter);
+        	reporterFactory.setHost("localhost");
+        	reporterFactory.setPort(testUdpPort);
+        	reporterFactory.setIncludesAttributes(includesAttributes);
+        	StatsdReporter reporter = reporterFactory.build(null);
+
+        	SortedMap<String, Meter> meters = new TreeMap<>();
+            Meter meter = new Meter();
+            meter.mark();
+            meter.mark(2);
+            meters.put("test-app-01.driver.TestMeter", meter);
+
+            reporter.report(new TreeMap<>(), new TreeMap<>(), new TreeMap<>(), meters, new TreeMap<>());
+
+            sleep(100);
+
+            String prefix = "spark.driver.testmeter.";
+            assertThatExists(server, prefix, "count", "3|g");
+            assertThatNotExists(server, prefix, "m1_rate", "0.00|ms");
+            assertThatNotExists(server, prefix, "m5_rate", "0.00|ms");
+            assertThatNotExists(server, prefix, "m15_rate", "0.00|ms");
+            // mean value depends on the timing and can't be predicted for the test
+            assertThatNotExists(server, prefix, "mean_rate", "|ms");
+            assertEquals(1, server.receivedMessages().size());
+
+        }
+    }
 
     private StatsdReporter buildTestReporter(int testUdpPort) {
-        return StatsdReporter
-                .forRegistry(null)
-                .formatter(formatter)
-                .host("localhost")
-                .port(testUdpPort)
-                .build();
+    	StatsdReporterFactory reporterFactory = new StatsdReporterFactory();
+    	reporterFactory.setFormatter(formatter);
+    	reporterFactory.setHost("localhost");
+    	reporterFactory.setPort(testUdpPort);
+        return reporterFactory.build(null);
```

Comment:
Hm, is this any different than what was previous besides verbosity?

Reply:
We have changed a way to create the `StatsdReporter` object, previously it was with a `Builder` inside it. Now it should be using `StatsdReporterFactory`.